libs/model/src/community/UpdateCommunity.command.ts

```diff
@@ -0,0 +1,160 @@
+      const snapshots = !snapshot
+        ? []
+        : typeof snapshot === 'string'
+          ? [snapshot]
+          : snapshot;
+      if (snapshots.length > 0 && community.base !== ChainBase.Ethereum)
+        throw new InvalidInput(UpdateCommunityErrors.SnapshotOnlyOnEthereum);
+
+      const newSpaces = snapshots.filter(
+        (s) => !community.snapshot_spaces.includes(s),
+      );
+      for (const space of newSpaces) {
+        if (!(await checkSnapshotObjectExists('space', space)))
+          throw new InvalidInput(UpdateCommunityErrors.SnapshotNotFound);
+      }
+
+      if (default_page && !has_homepage)
+        throw new InvalidInput(UpdateCommunityErrors.InvalidDefaultPage);
+
+      if (namespace) {
+        if (!transactionHash)
+          throw new InvalidInput(UpdateCommunityErrors.InvalidTransactionHash);
+
+        // we only permit the community admin and not the site admin to create namespace
+        if (actor.user.isAdmin)
+          throw new InvalidInput(UpdateCommunityErrors.NotAdmin);
+
+        community.namespace = namespace;
+        community.namespace_address =
+          await commonProtocol.newNamespaceValidator.validateNamespace(
+            namespace!,
+            transactionHash,
+            actor.address!,
+            community,
+          );
+      }
+
+      custom_domain && (community.custom_domain = custom_domain);
+      default_page && (community.default_page = default_page);
+      community.snapshot_spaces = snapshots;
+      name && (community.name = name);
+      description && (community.description = description);
+      default_symbol && (community.default_symbol = default_symbol);
+      icon_url && (community.icon_url = icon_url);
+      active !== undefined && (community.active = active);
+      type && (community.type = type);
+      stages_enabled !== undefined &&
+        (community.stages_enabled = stages_enabled);
+      has_homepage && (community.has_homepage = has_homepage);
+      chain_node_id && (community.chain_node_id = chain_node_id);
+      directory_page_enabled !== undefined &&
+        (community.directory_page_enabled = directory_page_enabled);
+      directory_page_chain_node_id !== undefined &&
+        (community.directory_page_chain_node_id = directory_page_chain_node_id);
+      default_summary_view !== undefined &&
+        (community.default_summary_view = default_summary_view);
+      social_links?.length && (community.social_links = social_links);
+      hide_projects !== undefined && (community.hide_projects = hide_projects);
+      custom_stages && (community.custom_stages = custom_stages);
+
+      await community.save();
+
+      // Suggested solution for serializing BigInts
+      // https://github.com/GoogleChromeLabs/jsbi/issues/30#issuecomment-1006086291
+      (BigInt.prototype as any).toJSON = function () {
+        return this.toString();
+      };
+      return (await models.Community.findOne({ where: { id } }))!.toJSON();
```

Comment:
Can we not just do `return community.toJSON()`?

Reply:
Honestly never seen this before but seems like something that should be set in top-level entry file so we have consistent parsing of BigInts. That said, I don't think we explicitly use native JS BigInts anywhere.

Reply:
Removed 👍